### PR TITLE
support broadcasting in _kl_categorical_categorical

### PR DESCRIPTION
**Pull request by joh4n**
Support broadcasting in _kl_categorical_categorical

this makes it possible to do:
```
import torch.distributions as dist
import torch
p_dist = dist.Categorical(torch.ones(1,10))
q_dist = dist.Categorical(torch.ones(100,10))
dist.kl_divergence(p_dist, q_dist)
```

### CODE REVIEW

**Comment by vishwakftw** [comment]
Could you please add a test in `test_distributions.py`?

**Comment by joh4n** [comment]
I added a basic test for it

File: test/test_distributions.py [hunk 1]
```diff
@@ -3014,6 +3014,7 @@ def __init__(self, probs):
             (binomial30, binomial30),
             (binomial_vectorized_count, binomial_vectorized_count),
             (categorical, categorical),
+            (Categorical(torch.ones(1, 10)), Categorical(torch.ones(3, 10))),
```
Comment (vishwakftw) [inline test/test_distributions.py]: Tuple members in this list are not instances of distributions, but a list of distributions. You could perhaps add a test in `self.infinite_examples` below, and ensure that the KL is indeed infinity.

**Review by vishwakftw — APPROVED** [review]
LGTM. Thank you for fixing this.

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**